### ISSUE TITLE
Updating shape_optimization.py for multi zone problems

### ISSUE THREAD

**Issue by MKUzuner9** [su2code/SU2#1279]
**Is your feature request related to a problem? Please describe.**
When using shape_optimization.py with MULTIZONE option enabled on v7.0.6, the script does not copy .cfg files in CONFIG_LIST setting.

**Describe the solution you'd like**
I am not too familiar with python so I am not sure about how hard this is to add but it sounds like a very simple addition. Therefore, if shape_optimization.py were to copy the .cfg files in CONFIG_LIST into direct solution directory which is named DIRECT, it would allow multi zone simulations to be run with shape_optimization.py as well.

**Comment by pcarruscag** [comment]
I am a little partial because I wrote it, but this https://github.com/su2code/FADO should do any kind of optimization you want.
Multiple operating points, manipulations of variables, running everything simultaneously... But I guess for something simple it is a little more work to setup.

**Comment by TobiKattmann** [comment]
I thought about tackling this but I will most likely also switch to FADO for these kinds of things for multizone (shape optimization, finite differences). 
Afaik there is a `pull` array which keeps track of all files that need to be copied, it might also be enough to add that to the `state` in `shaep_optimization.py`. Not to discourage you but I doubt that it will work with just that addition tbh. But you are more then welcome to give it a shot and see how it goes :+1:

**Comment by MKUzuner9** [comment]
Thanks for the replies and suggestions. I am trying to use built-in optimization abilities of SU2 for turbomachinery cases. So far the multi zone issue seems like one of the many problems. I realized this when using a single zone case while trying to avoid previous multi zone problem I encountered, when using v7.0.6, turbomachinery cases create a different kind of solution history file and it seems to interfere with the optimization process. Also, it seems like the optimization side of SU2 is created without considering turbomachinery simulations since the current objective functions are not really useful for turbomachinery optimization, and there are some objective functions that are in config_template.cfg but does not work such as SURFACE_MASSFLOW, SURFACE_TOTAL_PRESSURE and TORQUE. Therefore, it seems like the solution history and objective function problems need to be solved together in order to do optimization on turbomachinery cases with v7.0.6. Even though v7.0.6 turned out very satisfactory just for solving both single and multi zone turbomachinery cases I am trying to see if using an older version like v6.2.0 would change the situation on turbomachinery optimization. It may seem like I am avoiding going in and changing the code but I want to see first if a base version of SU2 can do the job.
